server/src/com/thoughtworks/go/config/GoFileConfigDataSource.java

```diff
@@ -332,14 +349,21 @@ private String getMergedConfig(NoOverwriteUpdateConfigCommand noOverwriteCommand
                 serverVersion.version(), timeProvider);
 
         String mergedConfigXml = configRepository.getConfigMergedWithLatestRevision(configRevision, oldMd5);
-        validateMergedXML(mergedConfigXml, latestMd5);
+        validateMergedXML(mergedConfigXml, latestMd5, partials);
+        cachedGoPartials.markAsValid(modifiedConfig.getPartials());
         return mergedConfigXml;
     }
 
-    private CruiseConfig validateMergedXML(String mergedConfigXml, String latestMd5) throws Exception {
+    //TODO: jyoti - write a test for a case with config-repo post-merge validation failing
```

Comment:
I think that is in `shouldNotAllowAGitMergeOfConcurrentChangesIfTheChangeCausesMergedPartialsToBecomeInvalid`?